crypto/o_time.c

```diff
@@ -358,3 +358,19 @@ static void julian_to_date(long jd, int *y, int *m, int *d)
     *m = j + 2 - (12 * L);
     *y = 100 * (n - 49) + i + L;
 }
+
+
+int OPENSSL_timegm(struct tm *tm, time_t *t)
+{
+    /* Figures it out without mucking the environment */
+    long basetime, currtime;
+
+    basetime = date_to_julian(1970, 1, 1);
+    currtime = date_to_julian(tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday);
+
+    if (t != NULL)
+        *t = (((currtime - basetime) * 24 + tm->tm_hour) * 60 + tm->tm_min) * 60 + tm->tm_sec;
```

Comment:
Side comment re "(but I do consider the non-POSIX VMS implementation to be a mistake)":

Not at the time, but it's true that by now, that implementation is mostly historical.  Back when it was implemented (before VMS 7.0), there was no gmtime(), or gmtime() was very badly coded (I don't recall which, just that pre-7.0 versions of the C RTL was...  not the best quality, and alternate implementation using true and well test VMS library calls were quite common at the time).

Thanks for the reminder, this might actually be a worthy piece to throw down the drain at some point.